### PR TITLE
Fix JSON.[STRLEN|OBJLEN] response inconsistently with Redis when the path doesn't exist

### PR DESCRIPTION
**Pull request by jackyyyyyssss**
5e7fd4968ee28ef848c9ac5160fe4b0
98ccdac6d5360ea5434d34345ff77c7
JSON.STRLEN and JSON.OBJLEN have path not return null

### CODE REVIEW

File: src/commands/cmd_json.cc [hunk 1]
```diff
@@ -559,9 +561,11 @@ class CommandJsonStrLen : public Commander {
 
     Optionals<uint64_t> results;
     auto s = json.StrLen(args_[1], path, &results);
-    if (s.IsNotFound()) {
+    if (s.IsNotFound() && args_.size() == 2) {
       *output = conn->NilString();
       return Status::OK();
+    } else if (s.IsNotFound() && args_.size() == 3) {
+      return {Status::RedisExecErr, "could not perform this operation on a key that doesn't exist"};
     }
```
Comment (git-hulk) [inline src/commands/cmd_json.cc]:
```suggestion
    if (s.IsNotFound()) {
      if (args_.size() == 2) {
        *output = conn->NilString();
        return Status::OK();
      } else {
        return {Status::RedisExecErr, "could not perform this operation on a key that doesn't exist"};
      }
    }
```

**Comment by jackyyyyyssss** [comment]
image
These errors seem to have nothing to do with the changes I made this time
please help retry

File: src/commands/cmd_json.cc [hunk 2]
```diff
@@ -382,9 +382,11 @@ class CommandJsonObjLen : public Commander {
 
     Optionals<uint64_t> results;
     auto s = json.ObjLen(args_[1], path, &results);
-    if (s.IsNotFound()) {
+    if (s.IsNotFound() && args_.size() == 2) {
       *output = conn->NilString();
       return Status::OK();
+    } else if (s.IsNotFound() && args_.size() == 3) {
+      return {Status::RedisExecErr, "Path '" + args_[2] + "' does not exist or not an object"};
     }
```
Comment (git-hulk) [inline src/commands/cmd_json.cc]:
```suggestion
    if (s.IsNotFound()) {
      if (args_.size() == 2) {
        *output = conn->NilString();
        return Status::OK();
      } else {
        return {Status::RedisExecErr, "Path '" + args_[2] + "' does not exist or not an object"};
      }
    }
```

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
@jackyyyyyssss Thanks for your fix.

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**